addons/purchase/models/purchase.py

```diff
@@ -413,6 +413,10 @@ def action_view_invoice(self):
             'default_company_id': self.company_id.id,
             'default_purchase_id': self.id,
         }
+        # Invoice_ids may be filtered depending on the user. To ensure we get all
+        # invoices related to the purchase order, we read them in sudo to fill the
+        # cache.
+        self.sudo()._read(['invoice_ids'])
```

Comment:
sudo() doesn't read in db. So if the cache doesn't have all the invoices, accessing them via sudo do not change anything

Reply:
Not sure I get the point. If the invoices were filtered for the current user, they will still be filtered by the record rules when opening the action. In other words, if the user has access to invoices `1,2,3` but there are in fact 2 extra (`4,5`), you will set the domain to `"[('id', 'in', " + str([1,2,3,4,5]) + ")]"`. But only `1,2,3` will be listed in the tree view since the record rules will be applied for the current user.

But anyway, I'm not against the change, no need to spend too much time on arguing on this.